### ISSUE TITLE
Fixme Windows Socket duplication

### ISSUE THREAD

**Issue by Wulfklaue** [denizzzka/dpq2#109]
Here is a nice error ( trying to connect to a external CockroachDB server ~ used PostgreSQL protocol ):

```
\Roaming\dub\packages\dpq2-1.0.0-rc.4\dpq2\src\dpq2\connection.d(161,13): Error: static assert:  "FIXME: implement socket duplication"
dmd failed with exit code 1.
```
Same issue with the 2.0 alpha's...

And:

```
/// Obtains duplicate file descriptor number of the connection socket to the server
    socket_t posixSocketDuplicate()
    {
        version(Windows)
        {
            static assert(false, "FIXME: implement socket duplication");
        }
        else // Posix OS
        {
            import core.sys.posix.unistd: dup;

            return cast(socket_t) dup(cast(socket_t) posixSocket);
        }
```

Not implemented in the code base.

**Comment by denizzzka** [comment]
(Related to #69 )

**Comment by Wulfklaue** [comment]
Been trying to get your code in my project but dub keeps reusing the old compiled version. So i manually added it to the package code and forced a new compilation by switching to dub release build. 

Same error on new build  ...

**Comment by denizzzka** [comment]
You code uses socket() ?

**Comment by Wulfklaue** [comment]
I simply use:

```
import vibe.db.postgresql; // "vibe-d-postgresql": "2.0.0-alpha.3"

void main()
{
	auto connectionInfo = "host=x.x.x.x port=26257 "
		~ "dbname=startrek  user=root sslmode=disable ";
	auto poolSize = 1;
	client = new PostgresClient(connectionInfo, poolSize);

       // ....

}

And vibe.db.postgresql depends on dpq2 ...

**Comment by denizzzka** [comment]
PR was changed, check it

**Comment by Wulfklaue** [comment]
Progress... This error is fixed.

But then we reach another issue down the line:

```
AppData\Roaming\dub\packages\dpq2-1.0.0-rc.4\dpq2\src\dpq2\conv\time.d(338,5): Error: static assert:  8u == 4u is false
AppData\Roaming\dub\packages\dpq2-1.0.0-rc.4\dpq2\src\dpq2\conv\time.d(112,12):        instantiated from here: rawTimeStamp2nativeTime!(TTimeStamp!true)
AppData\Roaming\dub\packages\dpq2-1.0.0-rc.4\dpq2\src\dpq2\conv\to_d_types.d(140,27):        instantiated from here: binaryValueAs!(TTimeStamp!true)
AppData\Roaming\dub\packages\dpq2-1.0.0-rc.4\dpq2\src\dpq2\conv\to_bson.d(150,24):        instantiated from here: tunnelForBinaryValueAsCalls!(TTimeStamp!true)
dmd failed with exit code 1.
```

**Comment by denizzzka** [comment]
wow. you are on 32 bit system?

**Comment by Wulfklaue** [comment]
People seem to forget that D on Windows by default only provides a 32bit version.

https://forum.dlang.org/thread/fiyfbeczwihhsbiudtaw@forum.dlang.org

**Comment by denizzzka** [comment]
#111
